pkg/jobs: remove un-needed check for `job_type` column existence

healthy-pod opened the pull request:
This code change makes the job creation code depend solely
on the version gate to check if `V23_1AddTypeColumnToJobsTable`
migration ran.

`TestTelemetry/schema` was rewritten.

Release note: None
Epic: [CRDB-21741](https://cockroachlabs.atlassian.net/browse/CRDB-21741)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98083)
<!-- Reviewable:end -->

Commit:
pkg/jobs: remove un-needed check for job_type column existence

This code change makes the job creation code depend solely
on the version gate to check if `V23_1AddTypeColumnToJobsTable`
migration ran.

`TestTelemetry/schema` was rewritten.

Release note: None
Epic: CRDB-21741

healthy-pod commented:
bors r=adityamaru
